django/api/services/calculate_rebate.py

```diff
@@ -0,0 +1,102 @@
+from ..models.go_electric_rebate_application import GoElectricRebateApplication
+from ..models.household_member import HouseholdMember
+from ..settings import INCOME_REBATES
+from enum import Enum
+
+
+class RebateType(Enum):
+    A = "A"
+    B = "B"
+    C = "C"
+    D = "Not Approved"
+
+
+def check_individual(primary_income):
+    if primary_income is None:
+        return RebateType.D.value
+    primary_income = int(primary_income)
+    if primary_income > INCOME_REBATES.get(RebateType.C.value).get("individual_income"):
+        return RebateType.D.value
+    elif primary_income <= INCOME_REBATES.get(RebateType.A.value).get(
+        "individual_income"
+    ):
+        return RebateType.A.value
+    elif primary_income <= INCOME_REBATES.get(RebateType.B.value).get(
+        "individual_income"
+    ):
+        return RebateType.B.value
+    elif primary_income <= INCOME_REBATES.get(RebateType.C.value).get(
+        "individual_income"
+    ):
+        return RebateType.C.value
+
+
+def check_household(primary_income, secondary_income):
+    if (primary_income is None) | (secondary_income is None):
+        return RebateType.D.value
+    household_income = int(primary_income) + int(secondary_income)
+    if household_income > INCOME_REBATES.get(RebateType.C.value).get(
+        "household_income"
+    ):
+        return RebateType.D.value
+    if household_income <= INCOME_REBATES.get(RebateType.A.value).get(
+        "household_income"
+    ):
+        return RebateType.A.value
+    elif household_income <= INCOME_REBATES.get(RebateType.B.value).get(
+        "household_income"
+    ):
+        return RebateType.B.value
+    elif household_income <= INCOME_REBATES.get(RebateType.C.value).get(
+        "household_income"
+    ):
+        return RebateType.C.value
+
+
+def get_final_rebate(individual_rebate, household_rebate):
+    if household_rebate == RebateType.A.value:
+        return INCOME_REBATES.get(RebateType.A.value).get("rebate")
+    if (
+        individual_rebate == RebateType.B.value
+        or household_rebate == RebateType.B.value
+    ):
+        return INCOME_REBATES.get(RebateType.B.value).get("rebate")
+    if (
+        individual_rebate == RebateType.C.value
+        or household_rebate == RebateType.C.value
+    ):
+        return INCOME_REBATES.get(RebateType.C.value).get("rebate")
+    if (
+        household_rebate == RebateType.D.value
+        and individual_rebate == RebateType.D.value
+    ):
+        return RebateType.D.value
+
+
+def calculate_rebate_amount(cra_response):
+    application_id = list(cra_response.keys())[0]
```

Comment:
the cra response can have many application ids. This can't just check the first one.

Reply:
`{
"B5t92XeH7NnFUwxc": [

{"sin": "302435839", "year": "2020", "income": "85687"}
],
"9uXLvNQS5vkKnscD": [
{"sin": "302435839", "year": "2020", "income": "85687"}
,
{"sin": "270300379", "year": "2020", "income": "81830"}
,
],
"ctW8gU57YX4xfQ9o": [

{"sin": "302435839", "year": "2020", "income": None}
],
}`

Reply:
I thought we were going to filter before it hits this function?